components/renku_data_services/app_config/logging.py

```diff
@@ -0,0 +1,312 @@
+            else:
+                kwargs["extra"] = self.extra
+            return f"[{rid}] {msg}", kwargs
+
+    @classmethod
+    def create(cls, logger: Logger, request_id: str) -> LoggerAdapter:
+        """Create a logger adapter that automatically adds the given `request_id` to each log message."""
+        return _RequestIdAdapter(logger, {"request_id": request_id})
+
+
+def configure_logging(cfg: Config = Config.from_env()) -> None:
+    """Configures logging library.
+
+    This should run before using a logger. It sets all loggers to
+    WARNING, except for our code that will log at INFO. Our code is
+    identified by the app root logger `renku_data_services`. All our
+    loggers should therefore be children of this logger.
+
+    Level for individual loggers can be overriden using the
+    `override_levels` argument. It is a map from logging level to a
+    list of logger names. The default reads it from environment
+    variables like `DEBUG_LOGGING=logger.name.one,logger.name.two`.
+    The pattern is `{LEVEL}_LOGGING` the value is a comma separated
+    list of logger names that will be configured to a minimum level of
+    `{LEVEL}`.
+
+    """
+    # To have a uniform format *everywhere*, there is only one
+    # handler. It is added to the root logger. However, imported
+    # modules may change this configuration at any time (and they do).
+    # This tries to remove all existing handlers as an best effort.
+    for ll in _Utils.get_all_loggers():
+        ll.setLevel(logging.NOTSET)
+        for hdl in ll.handlers:
+            ll.removeHandler(hdl)
+
+    handler = logging.StreamHandler()
+    handler.setFormatter(cfg.format_style.to_formatter())
+    logging.root.setLevel(cfg.root_level)
+    logging.root.addHandler(handler)
+    logging.getLogger(__app_root_logger).setLevel(cfg.app_level)
+
+    # this is for creating backwards compatibility, ideally these are
+    # defined as env vars in the specific process
+    logging.getLogger("sanic").setLevel(logging.INFO)
+    logging.getLogger("alembic").setLevel(logging.INFO)
+
+    logger = getLogger(__name__)
+
+    # override minimum level for specific loggers
+    for level, names in cfg.override_levels.items():
+        for name in names:
+            logger.info(f"Set threshold level: {name} -> {logging.getLevelName(level)}")
+            logging.getLogger(name).setLevel(level)
+
+
+def print_logger_setting(msg: str | None = None, show_all: bool = False) -> None:
+    """Prints the current logger settings.
+
+    It intentionally uses `print` to survive a messed up logger
+    config. It prints all loggers that have an explicit set level.
+    Others, like those with a `NOT_SET` level and the 'PlaceHolder'
+    loggers are not printed.
+
+    """
+    l_root = logging.Logger.root
+    output = ["================================================================="]
+    if msg is not None:
+        output.append(f"--- {msg} ---")
```

Comment:
just a hint, [str.center](https://docs.python.org/3/library/stdtypes.html#str.center) is a thing :slightly_smiling_face: though that might make it weird if msg is longer than the length. specified

Reply:
Ah thanks! This is a nice helper function. I only needed something to catch my eyes, so there is no assumption for terminal length or whatever :-)